### ISSUE TITLE
Feature: Support for Flux OCI features

### ISSUE THREAD

**Issue by kingdonb** [weaveworks/vscode-gitops-tools#309]
Flux has support for `HelmRepository` objects with `type: oci` since 0.31.0, released in June.

Flux also has a POC for the Kustomize controller supporting OCI, with a new `OCIRepository` kind that has only been published in pre-release. It is blocking on the reviews of fluxcd/flux2#2601. We need user and maintainer reviews of this feature to ensure that it is correct before the work on the final implementation can begin.

But for now, we have an implementation of this RFC:

* https://github.com/fluxcd/flux2/pull/2601

implemented as Proof of Concept in:

* https://github.com/fluxcd/kustomize-controller/pull/684
* https://github.com/fluxcd/source-controller/pull/788
* https://github.com/fluxcd/flux2/pull/2856

We should support the Helm OCI work that has already been released through the stable channel, and the `OCIRepository` work that is proposed through the prerelease channel of the VSCode extension. 👍 This feature is currently planned for target to release in 0.21.

**Comment by kingdonb** [comment]
Here is the current state (I have an `OCIRepository` which is not listed in Sources, and a Flux Kustomization which applies from it)

<img width="968" alt="Screen Shot 2022-07-02 at 8 52 10 AM" src="https://user-images.githubusercontent.com/3286998/177001651-b0f99e7b-b3c2-4cdb-bf8a-ba0fb74297df.png">



The `flux tree ks river-data-oci` looks like:

```
$ flux tree ks river-data-oci
Kustomization/flux-system/river-data-oci
├── Namespace/default
├── Service/default/river-data
├── Deployment/default/river-data
└── Ingress/default/river-data
```

It is probably one of these options, as all of the non-OCI things show fine what resources their tree has under them:

```
sourceRef:
  kind: OCIRepository
  name: river-data-oci
targetNamespace: default
```
<img width="1049" alt="Screen Shot 2022-07-02 at 8 54 10 AM" src="https://user-images.githubusercontent.com/3286998/177001715-ef18718d-d2e1-4f5e-b458-cc29a1794dc0.png">

It still reads the specs OK ☝️ 

I have not tested against Helm Controller's OCI-based repository source at all (that's next)

The experiment is being run out of here and here:
* https://github.com/juozasg/river-data-explorer/pull/2
* https://github.com/kingdonb/tailscale-k8s/tree/main/apps/worker/river-data-explorer

**Comment by kingdonb** [comment]
This has landed and is in 0.21.0 🎉
